openedx/openedx-learning
# System defined taxonomies

## Description

Subclasses related to build System-defined taxonomies:

- [Docs updated](https://github.com/openedx/openedx-learning/pull/67/files#diff-e20a5299480027adca1a87aac4dea6d00797403cc0a23aab43bf0204ddb3a069) to reflect the new architecture of taxonomies.
- `models.py` moved to `models/base.py`
- `tag_ref` [setter](https://github.com/openedx/openedx-learning/pull/67/files#diff-7ffb9100f9426c81c59e88e9beceb9e3a8089da152d1d605b7de97990e6f0d50R540) to allow override how to set a tag on an object_tag
- `object_tag_class` property [added to Taxonomy](https://github.com/openedx/openedx-learning/pull/67/files#diff-7ffb9100f9426c81c59e88e9beceb9e3a8089da152d1d605b7de97990e6f0d50R183)
- Base system-defined subclasses added.
- Language tags fixture.
- Taxonomy.system_defined updated to be a [hardcoded property](https://github.com/openedx/openedx-learning/pull/67/commits/1847248b48176014b989a25e03ec00b9dda5f26e#diff-7ffb9100f9426c81c59e88e9beceb9e3a8089da152d1d605b7de97990e6f0d50R203).

## Testing instructions

Manual testing is possible but not very useful, since we don't yet have views for these models and APIs.

But you can check that the added tests cover the expected polymorphism behavior supported by this change, and that tests are still covering 100% of the openedx_tagging module.


## Other information

Relates to https://github.com/openedx/modular-learning/issues/77
